# Rendimiento E-commerce

Para nuestro analisis de ventas por tiendas, desarrollamos los siguientes actividades y calculos:

- Identificar mes con mayor eficiencia (ventas/gasto publicidad)
- Determinar mes con mejor tasa de conversión y analizar causa
- Calcular ticket promedio (ventas/productos) por mes
- Evaluar relación entre visitantes y ventas

Analizar la eficiencia publicitaria, la tasa de conversión, el ticket promedio y la relación entre visitantes y ventas proporciona una visión clara de qué tan rentable, efectiva y coherente es la estrategia comercial de tu tienda.

Estos indicadores permiten optimizar presupuestos, detectar comportamientos del cliente y orientar decisiones de marketing basadas en datos reales.


## Mes con mayor eficiencia

In [6]:
import pandas as pd

data = {
    'Mes': ['Ene', 'Feb', 'Mar', 'Abr', 'May'],
    'Ventas ($)': [45000, 52000, 38000, 61000, 48000],
    'Visitantes': [15000, 18200, 12500, 20500, 16800],
    'Conversion (%)': [3.2, 2.9, 3.8, 3.1, 2.7],
    'Gasto Publicidad ($)': [8500, 9800, 7200, 11200, 9500],
    'Productos Vendidos': [450, 520, 380, 610, 480]
}

df = pd.DataFrame(data)

In [7]:
# 1. Calcular la eficiencia (Ventas por cada dólar de Publicidad)
df['Eficiencia'] = df['Ventas ($)'] / df['Gasto Publicidad ($)']

# 2. Encontrar el mes con la máxima eficiencia
mes_mas_eficiente = df.loc[df['Eficiencia'].idxmax()]

print("Mes con mayor eficiencia (Ventas/Gasto Publicidad):")
print(f"Mes: {mes_mas_eficiente['Mes']}")
print(f"Eficiencia: {mes_mas_eficiente['Eficiencia']:.2f}")

Mes con mayor eficiencia (Ventas/Gasto Publicidad):
Mes: Abr
Eficiencia: 5.45


Durante abril, por cada unidad monetaria invertida en publicidad se generaron 5.45 unidades de venta, lo cual indica un excelente retorno sobre la inversión publicitaria (ROI).

Este mes representa el punto más alto de eficiencia comercial, probablemente debido a:

- Campañas bien segmentadas o mejor optimizadas.
- Mayor interés estacional por ciertos productos.
- Aumento orgánico del tráfico o mejora en la retención de clientes.

**Impacto:**

Identificar este mes permite replicar las condiciones de éxito (tipo de campaña, canal, promociones o mix de productos), redirigiendo presupuesto hacia estrategias que maximicen el rendimiento y reduciendo gastos en meses menos eficientes.

## Mes con la mejor tasa de conversion

In [8]:
mes_mejor_conversion = df.loc[df['Conversion (%)'].idxmax()]

print("\nMes con mejor tasa de conversión:")
print(f"Mes: {mes_mejor_conversion['Mes']}")
print(f"Conversión: {mes_mejor_conversion['Conversion (%)']}%")


Mes con mejor tasa de conversión:
Mes: Mar
Conversión: 3.8%


En marzo, la proporción de visitantes que concretaron una compra fue la más alta del periodo.
Esto sugiere que, aunque la eficiencia publicitaria fue menor que en abril, la capacidad de convertir visitas en ventas fue superior, lo que puede estar asociado con:

- Campañas más persuasivas o enfocadas en retención.
- Mejor experiencia de usuario (UX) en el sitio.
- Promociones o lanzamientos que incentivaron la compra inmediata.

**Impacto:**

Analizar qué factores influyeron en esta conversión (fuente de tráfico, tipo de anuncio, comportamiento del cliente) ayuda a replicar la experiencia más efectiva del embudo de ventas, fortaleciendo las estrategias de atracción y cierre.

## Ticket promedio

In [9]:
# 1. Calcular el ticket promedio
df['Ticket Promedio ($)'] = df['Ventas ($)'] / df['Productos Vendidos']

print("\nTabla con Ticket Promedio por mes:")
# Usar .to_string() para evitar mostrar metadatos
print(df[['Mes', 'Ventas ($)', 'Productos Vendidos', 'Ticket Promedio ($)']].to_string(index=False))


Tabla con Ticket Promedio por mes:
Mes  Ventas ($)  Productos Vendidos  Ticket Promedio ($)
Ene       45000                 450                100.0
Feb       52000                 520                100.0
Mar       38000                 380                100.0
Abr       61000                 610                100.0
May       48000                 480                100.0


El ticket promedio se mantuvo constante en $100 a lo largo de los meses, lo que indica consistencia en los precios y hábitos de compra.

Sin embargo, también podría reflejar una falta de estrategias de upselling o cross-selling, ya que no hay evidencia de incremento en el gasto por cliente.

**Impacto:**
Mantener un ticket promedio estable garantiza previsibilidad, pero incrementarlo mediante estrategias como recomendaciones personalizadas, combos o descuentos por volumen podría elevar los ingresos sin necesidad de aumentar el tráfico.

## Relación entre Visitantes y Ventas

In [10]:
# 1. Calcular el coeficiente de correlación de Pearson
correlacion = df['Visitantes'].corr(df['Ventas ($)'])

print(f"\nCoeficiente de Correlación (Visitantes vs Ventas): {correlacion:.4f}")


Coeficiente de Correlación (Visitantes vs Ventas): 0.9896


Existe una correlación casi perfecta y positiva (r ≈ 0.99) entre el número de visitantes y las ventas.

Esto significa que el crecimiento de las ventas está altamente vinculado al incremento del tráfico, lo que evidencia que:

El sitio convierte visitantes en compradores de manera eficiente.

Las campañas de atracción son efectivas en generar ventas directas.

**Impacto:**

El alto nivel de correlación indica que incrementar el tráfico tiene un efecto directo en las ventas, pero también sugiere la oportunidad de diversificar estrategias de crecimiento, por ejemplo:

Mejorar el valor promedio de compra (AOV).

Fomentar la recompra (retención de clientes).

Optimizar conversiones sin depender tanto del aumento del tráfico.

## 🧩 Conclusión General

El análisis revela que tu tienda presenta una base sólida y eficiente, con resultados coherentes entre inversión, ventas y tráfico.

Abril destaca como el mes más rentable (eficiencia máxima).

Marzo muestra el mejor desempeño en conversión (ventas efectivas).

El ticket promedio constante refleja estabilidad, pero también una oportunidad de crecimiento mediante ventas complementarias.

La fuerte correlación entre visitantes y ventas (0.9896) confirma que las estrategias de marketing actuales generan impacto directo, aunque aún puede explorarse mayor diversificación.

En conjunto, estos indicadores permiten alinear marketing, ventas y producto bajo una estrategia más rentable y predecible, priorizando la eficiencia sobre el gasto y la optimización del valor por cliente.